### DISCUSSION TITLE
Documentation need update

### DISCUSSION THREAD

**Discussion by abdulkuddusa4** [flet-dev/flet#3454] [General]
Hello,
I'm new to this amazing framework. And I'm thankful to be part of this family.
Now, I see that the documentation is old(especially the tutorial section where people like me start their UI design journey).

issues:
- the documentation uses deprecated code 
- infact, the trello app code is giving error.
- And the documentation is not described enough.

My Proposal is to make the tutorial more documented and explain the components as we go along with the tutorial and introduce new components.

I can bring up Django documentation: https://docs.djangoproject.com/en/5.0/intro/tutorial01/

It has a section called "WRITE YOUR FIRST DJANGO APP"
So, We can make the tutorial section like this which will be designed to teach beginners.

This documentation is amazing for people from different backgrounds FLUTTER, Android, who know the UI design pattern.

But to a novice like me, I'm getting confused between NavigationDrawer and NavigationRail etc etc.

**Comment by baseplate-admin** [comment]
i am also thinking about this.

@FeodorFitsner, are you willing to move from [docusaurus](https://docusaurus.io/) to something like `sphinx` or `mkdocs`? I think that way we can document the python API efficiently.

**Reply by FeodorFitsner** [reply]
Oh, we've just got used to Docusaurus :) What can `sphinx` or `mkdocs` do which Docusaurus can't?

**Reply by baseplate-admin** [reply]
> What can sphinx or mkdocs do which Docusaurus can't?

They can facilitate the usage of [docstring](https://pypi.org/project/pydocstring/). Meaning the internal functions of flet gets automagically documented.

Here's an example of my project that has automatic documentation generation.

Python file : https://github.com/baseplate-admin/modern_colorthief/blob/d7a6314eba19197f1f12c76cf09a16309777cdc8/src/python/modern_colorthief/__init__.py#L10-L22

Documentation File: https://github.com/baseplate-admin/modern_colorthief/blob/d7a6314eba19197f1f12c76cf09a16309777cdc8/docs/api.md?plain=1#L3-L9

Documentation site : https://modern-colorthief.readthedocs.io/en/latest/api.html

**Reply by baseplate-admin** [reply]
[It can also be used to document dart](https://ncar.github.io/dart-developers-guide/documentation.html)
